extensions/data-plane-selector/data-plane-selector-client/src/test/java/org/eclipse/edc/connector/dataplane/selector/RemoteDataPlaneSelectorServiceTest.java

```diff
@@ -109,7 +113,17 @@ void find_withTransferType() {
         var result = service.select(DataAddress.Builder.newInstance().type("test1").build(), DataAddress.Builder.newInstance().type("test2").build(), "random", "transferType");
 
         assertThat(result).usingRecursiveComparison().isEqualTo(expected);
+    }
+
+    @Test
+    @Disabled // this will be tested when there will be a clear separation between the selector management and control api
```

Comment:
active code should be tested. The `addInstance` method has an impl, so why not test it?

Reply:
the problem is that this class is tested in a "bizzarre" way, by loading the "server" controller, that, currently, is a different one for this method (because of the separation between management/control api that will be solved in subsequent PRs). Anyway this logic is tested by the E2E tests currently.
Let's see if can I do something anyway